bigtable/client/cell.h

```diff
@@ -64,6 +76,12 @@ class Cell {
   /// this object is deleted.
   std::string const& value() const { return value_; }
 
+  /// Return the contents of this cell. The returned value is not valid after
+  /// this object is deleted.
+  std::uint64_t value_as_numeric() {
```

Comment:
The name does not quite capture what this function does.  `as_bigendian_int64`?

Reply:
We could introduce a `bigendian64_t` type, and then make this function look like this:

```C++
template <typename T>
bigendian64_t value_as<T>() const { ... }
```

Then we introduce a couple of helper traits:

```C++
namespace bigtable {
template <typename T> struct encoder {
  static T decode(std::string const& value);
  static std::string encode(T const& value);
};
```

Including a number of specializations:

```C++
template<> struct encode<bigendian64_t>  { /* stuff */ }
template<> struct encode<bigendian32_t> ..

// For protos, you have to do some complicated stuff with std::is_base<> to make sure Message derives from protobuf::Message
template<typename Message> struct encode<Message, ...> { ... };
}
```

 to convert anything to cell values, including protos, big endian numbers, and little endian numbers.  Anyway, that is a more complicated change, but we could create the skeleton in this PR and only support bigendian64_t for now?